File: server/models/notification.js
import mongoose from "mongoose";

const notificationSchema=new mongoose.Schema({
    title:String,
    description:String,
    users:[{
        type:mongoose.Types.ObjectId,
        ref:"User"
    }],
    createdAt:Date,
    readAt:Date,
    trash:Boolean
})

const Notification= mongoose.model("notifications",notificationSchema)
export default Notification;

File: client/team_track/lib/models/notification.dart


import 'package:team_track/models/user.dart';

class Notification {
  late String title;
  late String description;
  late List<User> users;
  late DateTime createdAt;
  late DateTime readAt;
  late bool trash;

  Notification({
    required this.title,
    required this.description,
    required this.users,
    required this.createdAt,
    required this.readAt,
    required this.trash,
  });
}